backend/project/endpoints/courses/courses.py

```diff
@@ -24,20 +29,47 @@
 class CourseForUser(Resource):
     """Api endpoint for the /courses link"""
 
-    @login_required
-    def get(self):
+    @login_required_return_uid
+    def get(self, uid=None):
         """ "
         Get function for /courses this will be the main endpoint
         to get all courses and filter by given query parameter like /courses?parameter=...
         parameters can be either one of the following: teacher,ufora_id,name.
         """
 
-        return query_selected_from_model(
-            Course,
-            RESPONSE_URL,
-            url_mapper={"course_id": RESPONSE_URL},
-            filters=request.args
-        )
+        try:
+            # Define the individual select statements
+            student_courses = select(Course).join(
+                CourseStudent, Course.course_id == CourseStudent.course_id).filter(
+                    CourseStudent.uid == uid)
+            admin_courses = select(Course).join(
+                CourseAdmin, Course.course_id == CourseAdmin.course_id).filter(
+                    CourseAdmin.uid == uid)
+            teacher_courses = select(Course).filter(Course.teacher == uid)
+
+            # Combine the select statements using union to remove duplicates
+            all_courses_query = union(student_courses, admin_courses, teacher_courses)
+
+            # Execute the union query and fetch all results as Course instances
+            courses = db.session.execute(all_courses_query).mappings().all()
+            courses_data = [dict(course) for course in courses]
+
+            for course in courses_data:
+                course["course_id"] = urljoin(f"{RESPONSE_URL}/", str(course['course_id']))
+
+            return {
+                "data": courses_data,
+                "url": RESPONSE_URL,
+                "message": "Courses fetched successfully"
+            }
+
+        except SQLAlchemyError as e:
+            print(e)
```

Comment:
Remove print statement, and the "... as e:" since e won't be used anymore.

Reply:
Also fixed in [4b6f170](https://github.com/SELab-2/UGent-3/pull/282/commits/4b6f1709669ce9a1f0cad404337d2ce7210d0578)